fluent-package/debian/rules

```diff
@@ -20,7 +20,7 @@ override_dh_auto_install:
 # related start/restart hook script embedding. It does not omit
 # deb-systemd-helper enable/update-state fluentd.service
 override_dh_installsystemd:
-	dh_installsystemd --no-stop-on-upgrade
+	dh_installsystemd --no-restart-after-upgrade --no-stop-on-upgrade
```

Comment:
`--no-restart-after-upgrade` option controls the result of the #DEBHELPER# macro expanding.
it effects `postinst` script in ubuntu focal package.

### before
Before adding `--no-restart-after-upgrade` option, #DEBHELPER# macro was expanded as following.
```sh
# Automatically added by dh_installsystemd/12.10ubuntu1                                                                                                                                                        
if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
        # In case this system is running systemd, we need to ensure that all                                                                                                                                   
        # necessary tmpfiles (if any) are created before starting.                                                                                                                                             
        if [ -d /run/systemd/system ] ; then
                systemd-tmpfiles --create fluentd.conf >/dev/null || true
        fi
fi
# End automatically added section                                                                                                                                                                              
# Automatically added by dh_installsystemd/12.10ubuntu1                                                                                                                                                        
if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
        # This will only remove masks created by d-s-h on package removal.                                                                                                                                     
        deb-systemd-helper unmask 'fluentd.service' >/dev/null || true

        # was-enabled defaults to true, so new installations run enable.                                                                                                                                       
        if deb-systemd-helper --quiet was-enabled 'fluentd.service'; then
                # Enables the unit on first installation, creates new                                                                                                                                          
                # symlinks on upgrades if the unit file has changed.                                                                                                                                           
                deb-systemd-helper enable 'fluentd.service' >/dev/null || true
        else
                # Update the statefile to add new symlinks (if any), which need to be                                                                                                                          
                # cleaned up on purge. Also remove old symlinks.                                                                                                                                               
                deb-systemd-helper update-state 'fluentd.service' >/dev/null || true
        fi
fi
# End automatically added section                                                                                                                                                                              
# Automatically added by dh_installsystemd/12.10ubuntu1                                                                                                                                                        
if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
        if [ -d /run/systemd/system ]; then
                systemctl --system daemon-reload >/dev/null || true
                if [ -n "$2" ]; then
                        _dh_action=restart
                else
                        _dh_action=start
                fi
                deb-systemd-invoke $_dh_action 'fluentd.service' >/dev/null || true
        fi
fi
# End automatically added section                                                                                                                                                                              ```

### after
After adding `--no-restart-after-upgrade` option, #DEBHELPER# macro was expanded as following.
```sh
# Automatically added by dh_installsystemd/12.10ubuntu1                                                                                                                                                        
if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
        # In case this system is running systemd, we need to ensure that all                                                                                                                                   
        # necessary tmpfiles (if any) are created before starting.                                                                                                                                             
        if [ -d /run/systemd/system ] ; then
                systemd-tmpfiles --create fluentd.conf >/dev/null || true
        fi
fi
# End automatically added section                                                                                                                                                                              
# Automatically added by dh_installsystemd/12.10ubuntu1                                                                                                                                                        
if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
        # This will only remove masks created by d-s-h on package removal.                                                                                                                                     
        deb-systemd-helper unmask 'fluentd.service' >/dev/null || true

        # was-enabled defaults to true, so new installations run enable.                                                                                                                                       
        if deb-systemd-helper --quiet was-enabled 'fluentd.service'; then
                # Enables the unit on first installation, creates new                                                                                                                                          
                # symlinks on upgrades if the unit file has changed.                                                                                                                                           
                deb-systemd-helper enable 'fluentd.service' >/dev/null || true
        else
                # Update the statefile to add new symlinks (if any), which need to be                                                                                                                          
                # cleaned up on purge. Also remove old symlinks.                                                                                                                                               
                deb-systemd-helper update-state 'fluentd.service' >/dev/null || true
        fi
fi
# End automatically added section                                                                                                                                                                              
# Automatically added by dh_installsystemd/12.10ubuntu1                                                                                                                                                        
if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
        if [ -d /run/systemd/system ]; then
                systemctl --system daemon-reload >/dev/null || true
                deb-systemd-invoke start 'fluentd.service' >/dev/null || true
        fi
fi
# End automatically added section                                                                                                                                                                              
```

### diff
```diff
root@jammy:~# diff before/DEBIAN/postinst after/DEBIAN/postinst 
246,251c246
<               if [ -n "$2" ]; then
<                       _dh_action=restart
<               else
<                       _dh_action=start
<               fi
<               deb-systemd-invoke $_dh_action 'fluentd.service' >/dev/null || true
---
>               deb-systemd-invoke start 'fluentd.service' >/dev/null || true
```

`--no-restart-after-upgrade` option will remove `restart` action.

Reply:
Thanks for investigating this mechanism!
Looks good to use `--no-restart-after-upgrade`.

Reply:
@kenhys Do you have any concerns?

Reply:
That's okay for me. 
Thanks.